### ISSUE TITLE
Error when use middleware with sockjs 

### ISSUE THREAD

**Issue by Crandel** [aio-libs/sockjs#23]
In your examples chat.py you don`t use middleware. But I try to create simple chat with authorization and it fails in middleware. When commented middleware functions, everything works fine
```python
TypeError: object generator can't be used in 'await' expression
```
Could you write example with middleware and for example with aiohttp_session

**Comment by fafhrd91** [comment]
you should post full traceback

**Comment by Crandel** [comment]
Okey
```python
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request POST /ws/037/fd0o4ang/xhr_streaming > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request GET /ws/037/k0tuslcr/eventsource > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request GET /ws/iframe.html > <class 'function'> <function SockJSRoute.iframe at 0x7f1522f06268> db_handler
<Request GET /ws/037/huop5nvj/eventsource > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request GET /ws/037/m5ptobdr/htmlfile > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request POST /ws/037/mtagmvxg/xhr > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request GET /ws/037/emv4c1ry/htmlfile > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request POST /ws/037/woalwafl/xhr > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
<Request GET /ws/037/1dgw4fdj/jsonp > <class 'method'> <bound method SockJSRoute.handler of <sockjs.route.SockJSRoute object at 0x7f15204a0e48>> db_handler
Error handling request
Traceback (most recent call last):
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/home/crandel/.virtualenvs/aiohttp/lib/python3.5/site-packages/aiohttp_session/__init__.py", line 129, in middleware
    response = yield from handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 37, in middleware
    return await handler(request)
  File "/home/crandel/work/projects/aiohttp/middlewares.py", line 20, in middleware
    response = await handler(request)
TypeError: object generator can't be used in 'await' expression
```
You can see my middleware here
[https://github.com/Crandel/aiohttp/blob/master/middlewares.py](url)

**Comment by garyvdm** [comment]
I've submitted a pull request that fixes this, but as a workaround you can change your middleware from py3.5 style coroutine to a py3.4 style coroutine: 

E.g. from this: 

    async def my_middleware(app, handler):

        @asyncio.coroutine
        def my_middleware_handler(request):
            return (yield from handler(request))

        return my_middleware_handler

To this:

    async def my_middleware(app, handler):

        async def my_middleware_handler(request):
            return (await handler(request))

        return my_middleware_handler

**Comment by asvetlov** [comment]
Fixed by #25
